base/test.jl

```diff
@@ -908,4 +909,16 @@ function detect_ambiguities(mods...; imported::Bool=false)
     collect(ambs)
 end
 
+"""
+The `GenericString` can be used to test generic string APIs that program to
+the `AbstractString` interface. In order to ensure that functions can work
+with string types besides the standard `String` type.
+"""
+immutable GenericString <: AbstractString
```

Comment:
We do actually install that file to `joinpath(JULIA_HOME, Base.DATAROOTDIR, "julia", "test", "TestHelpers.jl")` so that would work. It's a little bit hidden, but might make more sense than here.